django/contrib/admin/static/admin/js/nav_sidebar.js

```diff
@@ -2,18 +2,37 @@
 {
     const toggleNavSidebar = document.getElementById('toggle-nav-sidebar');
     if (toggleNavSidebar !== null) {
+        const navLinks = document.querySelectorAll('#nav-sidebar a');
+
+        function disableNavLinkTabbing() {
+            for (const navLink of navLinks) {
+                navLink.setAttribute("tabindex", "-1");
+            }
+        }
+
+        function enableNavLinkTabbing() {
+            for (const navLink of navLinks) {
+                navLink.setAttribute("tabindex", "0");
+            }
+        }
+
         const main = document.getElementById('main');
         let navSidebarIsOpen = localStorage.getItem('django.admin.navSidebarIsOpen');
         if (navSidebarIsOpen === null) {
             navSidebarIsOpen = 'true';
         }
+        if (navSidebarIsOpen === 'false') {
+            disableNavLinkTabbing();
+        }
         main.classList.toggle('shifted', navSidebarIsOpen === 'true');
 
         toggleNavSidebar.addEventListener('click', function() {
-            if (navSidebarIsOpen == 'true') {
+            if (navSidebarIsOpen === 'true') {
```

Comment:
We should enable the eslint rule "eqeqeq" to let the tools catch `==` misuse. I've done this in #12922.